### PR TITLE
[Bugfix] Thread LTX-2 linear prefixes for ModelOpt NVFP4

### PR DESCRIPTION
**Pull request by Qyx5-5**
## Motivation

Current SGLang main contains the ModelOpt NVFP4 scale-layout conversion for
non-packed-QKV DiTs, but LTX-2 does not propagate checkpoint-compatible layer
prefixes into its quantizable parallel-linear modules.

This breaks two coupled contracts:

1. `ModelOptQuantConfig.is_layer_excluded()` cannot match checkpoint exclusion
   patterns such as `transformer_blocks.0.*`.
2. `uses_flux1_scale_layout` cannot recognize the affected LTX-2 layers because
   it requires a prefix beginning with `transformer_blocks.` or
   `single_transformer_blocks.`.

The resulting load-time packed-versus-dense shape assertion can look like a
checkpoint architecture mismatch, but the serialized `(2048, 1024)` FP4
weight is the packed representation of a logical `(2048, 2048)` matrix.

Sibling DiTs including Flux, Flux2, GLM-Image, Qwen-Image, and the SANA-WM
refiner already construct their blocks with `transformer_blocks.<index>`.

## Modifications

- Construct each LTX-2 block with `transformer_blocks.<index>`.
- Propagate prefixes through attention `to_q`, `to_k`, `to_v`,
  `to_gate_logits`, and `to_out.0`.
- Propagate prefixes through feed-forward `proj_in` and `proj_out` and the
  block `ff` and `audio_ff` paths.
- Add stable prefixes to `patchify_proj`, `audio_patchify_proj`, `proj_out`,
  and `audio_proj_out`.
- Add a CPU-only regression test that constructs a tiny two-block LTX-2 model
  and verifies every quantized parallel linear receives its exact checkpoint
  module path.

The patch changes `ltx_2.py` and one focused unit-test file. It does not modify
checkpoint bytes, quantization math, or backend selection.

## Accuracy Tests

Validated on upstream main
`61057bda6c4b8cda1117d74f100d0735645c0cfb` at commit
`3528d293bda89c748da7fe559f97430b4c238ef2`.

Static checks:

```bash
uvx --from pre-commit pre-commit run --files \
  python/sglang/multimodal_gen/runtime/models/dits/ltx_2.py \
  python/sglang/multimodal_gen/test/unit/test_ltx2_quantized_linear_prefixes.py
python -m py_compile <both changed Python files>
git diff --check origin/main...HEAD
```

All applicable hooks passed.

Focused regression and the exact diffusion unit-suite entry point:

```bash
python -m pytest -q \
  python/sglang/multimodal_gen/test/unit/test_ltx2_quantized_linear_prefixes.py

cd python
SGLANG_DIFFUSION_CONSISTENCY_PLATFORM=h100 \
  python3 sglang/multimodal_gen/test/run_suite.py --suite unit
```

- Focused regression: `1 passed`.
- Diffusion unit suite: `797 passed, 4 skipped, 50 subtests passed`.
- The regression verifies 72 quantized parallel-linear instances across a
  two-block gated-attention configuration, including both block indices and
  the four top-level projections.

Full end-to-end workload:

- `LTX2TwoStagePipeline`
- felt-birds prompt, seed 10, guidance 3
- 768x1280, 241 frames, 24 fps, 30 steps, audio
- Cache-DiT and SageAttention2
- one fresh request per mode, serialized on one SM120 GPU

| Mode | Result | Inference | Reported peak memory |
| --- | --- | ---: | ---: |
| BF16 | valid coherent H.264 + AAC | 60.090 s | 77,756 MB |
| W4A4 NVFP4 | valid coherent H.264 + AAC | 48.011 s | 55,494 MB |

The current native loader reported:

```text
group_size=16
excluded modules=1010
packed_qkv=False
scale_layout=linear
swap_nibbles=False
```

Five-frame inspection at frames 0/60/120/180/240 found coherent felt birds,
a cardboard/twine birdhouse, stable motion, and no visual collapse in W4A4.
Different BF16 and W4A4 compositions are expected trajectory drift and are
not treated as a pixel-parity failure.

### Visual evidence

BF16 is on the left and W4A4 NVFP4 is on the right. Columns show frames
0/60/120/180/240.

<details>
<summary>Full 10-second BF16 versus W4A4 side-by-side video</summary>

</details>

Both processes released the GPU to 0 MiB with volatile ECC 0/0.

## Speed Tests and Profiling

No averaged speed claim is made. The table reports one fresh full request per
mode for validation context.

The W4A4 run explicitly used the CUTLASS FlashInfer backend because current
diffusion backend selection defaults to TRTLLM, which the recorded FlashInfer
build rejects on SM120. Backend selection is a separate issue and is outside
this prefix-only change.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
  - All hooks applicable to the two changed Python files passed.
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
  - Added `test_ltx2_quantized_linear_prefixes.py`; the complete diffusion unit
    suite passed with the new test collected.
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
  - Not applicable: this restores internal module-name propagation and does
    not change a user-facing interface.
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
  - Full-resolution BF16 and W4A4 results are above. Timings are disclosed as
    single fresh requests; this PR makes no speed claim.
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

## AI assistance disclosure

AI tooling assisted with investigation, implementation review, experiment
orchestration, testing, and drafting. The patch and PR are reviewed and
submitted by @Qyx5-5.

---
### CI States

Latest PR Test (Base): :x: [Run #31019611162](https://github.com/sgl-project/sglang/actions/runs/31019611162)
Latest PR Test (Extra): :x: [Run #31019611225](https://github.com/sgl-project/sglang/actions/runs/31019611225)

### CODE REVIEW

**Comment by Qyx5-5** [comment]
@mickqian @BBuf Could one of you please add the `run-ci` label for this diffusion PR? The focused regression passed, and the complete diffusion unit suite passed locally (`797 passed, 4 skipped, 50 subtests passed`) on the submitted SHA. Full-resolution BF16/W4A4 validation details are included in the PR body. Thank you.

**Comment by Qyx5-5** [comment]
Rebased this PR onto current `main` (`b3cdd016baeb`), which includes the active `MIN_BASE_SHA` (`cdff33d738d5`).

Validation completed:
- `git range-diff` shows the patch is unchanged (one commit, two files).
- `git diff --check` and Python compilation passed for both changed files.
- All applicable two-file pre-commit hooks passed.
- The focused LTX-2 prefix regression passed (72 quantized linear prefixes).

Could an authorized maintainer please trigger `/tag-and-rerun-ci`?
